include/drivers/video.h

```diff
@@ -0,0 +1,317 @@
+/**
+ * @file
+ *
+ * @brief Public APIs for Video.
+ */
+
+/*
+ * Copyright (c) 2019 Linaro Limited.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_VIDEO_H_
+#define ZEPHYR_INCLUDE_VIDEO_H_
+
+#include <device.h>
+#include <stddef.h>
+#include <zephyr.h>
+
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum video_device_type {
+	video_CAPTURE_DEVICE,
+	video_OUTPUT_DEVICE,
+	video_CONTROL_DEVICE,
+};
+
+struct video_device_config {
+	enum video_device_type type;
+};
+
+struct video_format {
+	u32_t pixelformat;
+	u32_t width;
+	u32_t height;
+};
+
+struct video_buffer {
+	/** Reserved for video driver, keep it first (e.g. fifo) */
+	void *driver_data;
+
+	/** frame buffer */
+	void *frame;
+
+	/** Length of frame buffer in bytes */
+	size_t len;
+
+	/** Flags for the buffer */
+	u8_t flags;
+
+	/** format **/
+	struct video_format fmt;
+};
+
+/**
+ * @typedef video_api_set_format_t
+ * @brief Set video format
+ * See @a video_api_set_fmt() for argument.
+ * description.
+ */
+typedef int (*video_api_set_format_t)(struct device *dev,
+				      struct video_format *fmt);
+
+/**
+ * @typedef video_api_get_format_t
+ * @brief get current video format
+ * See @a video_api_set_fmt() for argument.
```

Comment:
`video_api_set_fmt()` to `video_api_get_fmt()`

Reply:
thanks fixed.